examples/tutorials/introduction/4_properties.py

```diff
@@ -0,0 +1,206 @@
+"""
+Property Traits
+===============
+
+The ``Image`` class has three traits which are closely related: ``scan_size``,
+``scan_width`` and ``scan_height``.  We would ideally like to keep all of these
+synchronised.  This can be done with trait observation, as shown in the previous
+section, but this sort of pattern is common enough that Traits has some in-built
+helpers.
+
+Instead of declaring that ``scan_width = Float()`` we could instead declare it
+to be a ``Property`` trait type.  Property traits are similar to ``@property``
+decorators in standard Python in that rather than storing a value, they compute
+a derived value via a "getter", and optionally store values via a "setter".
+Traits uses specially named methods of the form ``_get_<property>`` and
+``_set_property`` for these "getters" and "setters."  Additionally, we need to
+know when the value of the property might change, and so we need to declare
+what traits to listen to to know when the property might change.
+
+What this means is that we can define ``scan_width`` as a property by::
+
+    class Image(HasTraits):
+        ...
+
+        scan_width = Property(Float, depends_on='scan_size')
+
+        def _get_scan_width(self):
+            return self.scan_size[0]
+
+        def _set_scan_width(self, value):
+            self.scan_size = (value, self.scan_height)
+
+Traits will then take care of hooking up all the required observers to make
+everything work as expected.
+
+In this example we wrote both a "getter" and a "setter", but in many cases all
+you need and want is a "getter", and this is OK for Traits.  Additionally,
```

Comment:
Maybe it is already obvious: should we mention that if there is only a getter, the property cannot be set (i.e. readonly)?